src/components/ObsDetails/ObsDetails.js

```diff
@@ -159,6 +158,19 @@ const ObsDetails = ( ): Node => {
     }
   }, [observation, localObservation, realm, markViewedMutation, uuid] );
 
+  useEffect( ( ) => {
+    const obsCreatedLocally = observation?.id === null;
+    const obsOwnedByCurrentUser = observation?.user?.id === currentUser?.id;
+
+    const navToObsEdit = ( ) => navigation.navigate( "ObsEdit", { uuid: observation?.uuid } );
+    const editIcon = ( ) => ( obsCreatedLocally || obsOwnedByCurrentUser )
+    && <IconButton icon="pencil" onPress={navToObsEdit} textColor={colors.gray} />;
+
+    navigation.setOptions( {
+      headerRight: editIcon
+    } );
+  }, [navigation, observation, currentUser] );
+
```

Comment:
Just so I understand, this is in an effect b/c `currentUser` gets fetched asynchronously? And possibly `observation` if it's remote?

Reply:
Not exactly... this is in an effect because that's the [recommended way](https://reactnavigation.org/docs/header-buttons/#header-interaction-with-its-screen-component) to handle interactions between a component and a react navigation header.

In this case, the component knows there is an observation with a specific uuid, and if that observation belongs to the current user we want to show the edit button in the header. The header itself doesn't have a way of knowing which observation is being displayed in the component (& whether that observation belongs to the current user), so we need to use `navigation.setOptions` here to pass the correct right header icon from the component to the header.